src/main/java/mate/academy/onlinebookstore/repository/book/spec/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.onlinebookstore.repository.book.spec;
+
+import java.util.Arrays;
+import mate.academy.onlinebookstore.model.Book;
+import mate.academy.onlinebookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String KEY = "author";
```

Comment:
```suggestion
    private static final String AUTHOR_COLUMN_NAME = "author";
```

same for other providers